src/Workspace.cpp

```diff
@@ -195,15 +198,13 @@ bool WorkspaceFolder::updateSourceMap()
         frontend.clear();
         fileResolver.updateSourceMap(sourceMapContents.value());
 
-        // Recreate instance types
         auto config = client->getConfiguration(rootUri);
-        instanceTypes.clear();
         // NOTE: expressive types is always enabled for autocomplete, regardless of the setting!
         // We pass the same setting even when we are registering autocomplete globals since
         // the setting impacts what happens to diagnostics (as both calls overwrite frontend.prepareModuleScope)
-        types::registerInstanceTypes(frontend, frontend.globals, instanceTypes, fileResolver,
+        platform->handleSourcemapUpdate(frontend, frontend.globals, fileResolver,
```

Comment:
My issue with abstracting away sourcemaps was that they seem to be quite tightly integrated with the main LSP, especially in `WorkspaceFileResolver`. Should the concept of `SourceNode`, virtual paths, etc. also be separated from the main LSP? Alternatively, is it possible for the concept of a sourcemap to be used by other platforms?

Reply:
Ah yeah, that is problematic. WorkspaceFileResolver built up a lot of cruft.

I wonder if we can generalise the idea of virtual and file paths, where "virtual" is treated as "LSP internal" in a sense. Platform should then only have two public API: a way to convert from file path to virtual, and then from virtual to real. The concept of "SourceNode" would then be Roblox-specific.

Or maybe it would even just be simpler for Platforms to implement their own file resolvers as a whole. FileResolvers are quite platform-specific.

I'm not sure how this would look without trying it out though. We could defer it to a future commit to prevent too much changing in one PR.